static/src/js/components/TemporalDisplay/__tests__/TemporalSelectionDropdown.test.js

```diff
@@ -1,147 +1,216 @@
 import React from 'react'
-import Enzyme, { shallow } from 'enzyme'
-import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
-import moment from 'moment'
+import ReactDOM from 'react-dom'
+import {
+  fireEvent,
+  render,
+  screen,
+  waitFor
+} from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import '@testing-library/jest-dom'
 
-import Dropdown from 'react-bootstrap/Dropdown'
+import moment from 'moment'
 
 import TemporalSelectionDropdown from '../TemporalSelectionDropdown'
 
-Enzyme.configure({ adapter: new Adapter() })
-
-function setup() {
+const setup = (overrideProps) => {
   const props = {
-    temporalSearch: {},
-    onChangeQuery: jest.fn()
+    temporalSearch: {
+      endDate: '',
+      startDate: '',
+      isRecurring: false
+    },
+    onChangeQuery: jest.fn(),
+    ...overrideProps
   }
 
-  const enzymeWrapper = shallow(<TemporalSelectionDropdown {...props} />)
-
-  return {
-    enzymeWrapper,
-    props
-  }
+  return render(
+    <TemporalSelectionDropdown {...props} />
+  )
 }
 
 describe('TemporalSelectionDropdown component', () => {
-  test('on load should be closed on inital render', () => {
-    const { enzymeWrapper } = setup()
-    expect(enzymeWrapper.find(Dropdown).prop('show')).toBe(false)
+  beforeAll(() => {
+    ReactDOM.createPortal = jest.fn((dropdown) => dropdown)
+  })
+
+  afterEach(() => {
+    ReactDOM.createPortal.mockClear()
   })
 
-  test('when clicked toggles the state of show ', () => {
-    const { enzymeWrapper } = setup()
+  test('on load should be closed on inital render', () => {
+    setup()
 
-    enzymeWrapper.instance().onToggleClick()
-    expect(enzymeWrapper.find(Dropdown).prop('show')).toBe(true)
+    expect(screen.queryByRole('button')).toBeInTheDocument()
+    expect(screen.queryByLabelText('Start')).not.toBeInTheDocument()
   })
 
-  test('sets the start date correctly when an valid date is passed', () => {
-    const { enzymeWrapper } = setup()
-    const testObj = moment.utc('2012-01-01 12:00:00', 'YYYY-MM-DD HH:mm:ss', true)
+  test('when clicked toggles the state of show ', async () => {
+    const user = userEvent.setup()
+    const { unmount } = setup({})
+
+    const btn = screen.getByRole('button')
+    expect(btn).toBeInTheDocument()
+
+    await waitFor(async () => {
+      await user.click(btn)
+    })
+
+    const startLabel = screen.getByText(/Start/i)
 
-    enzymeWrapper.instance().setStartDate(testObj)
-    expect(enzymeWrapper.state().temporal.startDate).toEqual('2012-01-01T12:00:00.000Z')
+    expect(startLabel).toBeInTheDocument()
+    unmount()
```

Comment:
did you need this `unmount` for the test because something we being kept between them?